src/repl/scala/tools/nsc/MainGenericRunner.scala

```diff
@@ -101,5 +101,5 @@ class MainGenericRunner {
 }
 
 object MainGenericRunner extends MainGenericRunner {
-  def main(args: Array[String]): Unit = if (!process(args)) sys.exit(1)
+  def main(args: Array[String]): Unit = System.exit(if (process(args)) 0 else 1)
```

Comment:
This changes behavior.. if `process` is true, we should not call `exit`. Otherwise this will kill the application without waiting for spawned non-daemon threads to finish. https://github.com/scala/scala/blob/v2.12.1/test/files/run/lazy-concurrent.scala is an example.